tests/integration/test_examples.py

```diff
@@ -50,279 +50,117 @@ def example_runner(examples_dir):
-    mlflow_runs = mlflow.search_runs(
-        experiment_ids=[mlflow_experiment.experiment_id], output_format="list"
-    )
-    assert len(mlflow_runs) == 2
-
-    # fetch the MLflow run created for the first pipeline run
-    mlflow_runs = mlflow.search_runs(
-        experiment_ids=[mlflow_experiment.experiment_id],
-        filter_string=f'tags.mlflow.runName = "{first_run.name}"',
-        output_format="list",
-    )
-    assert len(mlflow_runs) == 1
-    first_mlflow_run = mlflow_runs[0]
-
-    # fetch the MLflow run created for the second pipeline run
-    mlflow_runs = mlflow.search_runs(
-        experiment_ids=[mlflow_experiment.experiment_id],
-        filter_string=f'tags.mlflow.runName = "{second_run.name}"',
-        output_format="list",
-    )
-    assert len(mlflow_runs) == 1
-    second_mlflow_run = mlflow_runs[0]
 
-    client = MlflowClient()
-    # fetch the MLflow artifacts logged during the first pipeline run
-    artifacts = client.list_artifacts(first_mlflow_run.info.run_id)
-    assert len(artifacts) == 3
-
-    # fetch the MLflow artifacts logged during the second pipeline run
-    artifacts = client.list_artifacts(second_mlflow_run.info.run_id)
-    assert len(artifacts) == 3
-
-
-def test_whylogs_profiling(examples_dir: Path):
-    """Testing the functionality of the whylogs example
+    name: str
+    validation_function: ExampleValidationFunction
+
+
+examples = [
+    ExampleIntegrationTestConfiguration(
+        name="quickstart",
+        validation_function=generate_basic_validation_function(
+            pipeline_name="mnist_pipeline", step_count=3
+        ),
+    ),
+    ExampleIntegrationTestConfiguration(
+        name="not_so_quickstart",
+        validation_function=generate_basic_validation_function(
+            pipeline_name="mnist_pipeline", step_count=4, run_count=3
+        ),
+    ),
+    ExampleIntegrationTestConfiguration(
+        name="caching", validation_function=caching_example_validation
+    ),
+    ExampleIntegrationTestConfiguration(
+        name="custom_materializer",
+        validation_function=generate_basic_validation_function(
+            pipeline_name="pipe", step_count=2
+        ),
+    ),
+    ExampleIntegrationTestConfiguration(
+        name="fetch_historical_runs",
+        validation_function=generate_basic_validation_function(
+            pipeline_name="mnist_pipeline", step_count=3
+        ),
+    ),
+    ExampleIntegrationTestConfiguration(
+        name="kubeflow",
+        validation_function=generate_basic_validation_function(
+            pipeline_name="mnist_pipeline", step_count=4
+        ),
+    ),
+    ExampleIntegrationTestConfiguration(
+        name="drift_detection",
+        validation_function=drift_detection_example_validation,
+    ),
+    ExampleIntegrationTestConfiguration(
+        name="mlflow_tracking",
+        validation_function=mlflow_tracking_example_validation,
+    ),
+    # TODO [HIGH]: Enable running the whylogs example on kubeflow
```

Comment:
Is this because the whylogs example is currently failing on Kubeflow ?

Reply:
Exactly, but as long as we're not running on kubeflow I think this can be re-enabled